aws-iot-softwarepackage/src/main/java/software/amazon/iot/softwarepackage/CreateHandler.java

```diff
@@ -0,0 +1,188 @@
+    private static final int MAX_PACKAGE_NAME_LENGTH = 128;
+    private Logger logger;
+    private String clientToken;
+
+    @Override
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<IotClient> proxyClient,
+            final Logger logger) {
+
+        this.logger = logger;
+        this.clientToken = request.getClientRequestToken();
+
+        final ResourceModel resourceModel = request.getDesiredResourceState();
+
+        // create a package name if not provided by user
+        if (StringUtils.isNullOrEmpty(resourceModel.getPackageName())) {
+            resourceModel.setPackageName(generateName(request));
+        }
+
+        resourceModel.setDefaultVersionName(DEFAULT_PACKAGE_VERSION_NAME);
+
+        return ProgressEvent.progress(resourceModel, callbackContext)
+                .then(progress ->
+                        proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                                .translateToServiceRequest(Translator::translateToUpdateFIRequest)
+                                .makeServiceCall(this::updateIndexingConfiguration)
+                                .progress())
+                .then(progress ->
+                        proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                                .translateToServiceRequest(Translator::translateToCreateRequest)
+                                .makeServiceCall(this::createResource)
+                                .progress())
+                .then(progress ->
+                        proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                                .translateToServiceRequest(Translator::translateToCreateRequestForPackageVersion)
+                                .makeServiceCall(this::createResourceForPackageVersion)
+                                .progress())
+                .then(progress ->
+                        proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                                .translateToServiceRequest(Translator::translateToUpdateRequestForPackageVersion)
+                                .makeServiceCall(this::updateResourceForPackageVersion)
+                                .progress())
+                .then(progress ->
+                        proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                                .translateToServiceRequest(Translator::translateToUpdateRequest)
+                                .makeServiceCall(this::updateResourceForDefaultVersionName)
+                                .progress())
+                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
```

Comment:
Could you give me a brief summary what we do in create handler

Reply:
this part has been simplified as of the latest commit